src/monitors/turbFlowStats.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef ABLATELIBRARY_TURBFLOWSTATS_HPP
+#define ABLATELIBRARY_TURBFLOWSTATS_HPP
+
+#include <string>
+#include "domain/field.hpp"
+#include "eos/eos.hpp"
+#include "io/interval/interval.hpp"
+#include "io/serializable.hpp"
+#include "monitor.hpp"
+#include "monitors/logs/log.hpp"
+
+namespace ablate::monitors {
+
+class TurbFlowStats : public Monitor, public io::Serializable {
+    using ttf = ablate::eos::ThermodynamicFunction;
+
+   protected:
+    void AddField(DM& dm, const char* nameField, PetscInt numComp);
+
+   private:
+    const std::vector<std::string> fieldNames;
+    const std::shared_ptr<ablate::eos::EOS> eos;
+    const std::shared_ptr<io::interval::Interval> interval;
+    const std::string name = "TurbFlowStats";
+    std::vector<PetscInt> fieldTrack;
+    std::vector<PetscInt> fieldComps;
+    Vec turbVec;
+    DM turbDM;
+    ttf densityFunc;
+
+    // Keeps track of the categories used
+    struct {
+        PetscInt densitySum = 0;
+        PetscInt densityDtSum = 1;
+        PetscInt densityMult = 2;
+        PetscInt densityDtMult = 3;
+        PetscInt densitySqr = 4;
+        PetscInt sum = 5;
+        PetscInt sumSqr = 6;
+        PetscInt favreAvg = 7;
+        PetscInt rms = 8;
+        PetscInt mRms = 9;
+
+        void SetMode(PetscInt numComps) {
+            densityMult = 2;
+            densityDtMult = densityMult + numComps;
+            densitySqr = densityDtMult + numComps;
+            sum = densitySqr + numComps;
+            sumSqr = sum + numComps;
+            favreAvg = sumSqr + numComps;
+            rms = favreAvg + numComps;
+            mRms = rms + numComps;
+        }
+    } CatOffset, CatOrder;
+
+    inline static const double tiny = 1e-30;
```

Comment:
@owenmylotte added a constants class in the utilities folder.  Would you mind moving tiny to this class and calling it from there?